test_conformance/api/test_kernel_private_memory_size.cpp

```diff
@@ -0,0 +1,33 @@
+#include "harness/errorHelpers.h"
+#include "harness/typeWrappers.h"
+#include <iostream>
+#include "procs.h"
+
+int test_kernel_private_memory_size(cl_device_id deviceID, cl_context context,
+                                    cl_command_queue queue, int num_elements)
+{
+    const char* TEST_KERNEL =
+        R"(__kernel void private_memory( __global ulong *buffer ){
+         volatile __private ulong x[1];
+         buffer[0] = x[0];
+         })";
+
+    cl_int err = 0;
+    clProgramWrapper program(
+        clCreateProgramWithSource(context, 1, &TEST_KERNEL, nullptr, &err));
+    ASSERT_SUCCESS(err, "clCreateProgramWithSource");
+    err = clBuildProgram(program, 1, &deviceID, nullptr, nullptr, nullptr);
+    ASSERT_SUCCESS(err, "clBuildProgram");
+
+    /* Create the Kernel*/
+    clKernelWrapper kernel(clCreateKernel(program, "private_memory", &err));
+    ASSERT_SUCCESS(err, "clCreateKernel");
```

Comment:
Is there a reason this test uses `ASSERT_ERROR` instead of `test_error`?  I think most tests use `test_error`, which will simply return a failure vs. throwing an exception.

Reply:
Not really, I'm still new and learning the about the CTS. Thank you for pointing this out to me. I've updated this in the latest patch.